Sources/Core/Model/Attachment.swift

```diff
@@ -211,9 +212,78 @@ public extension Attachment {
 }
 
 /// An attachment type.
-public enum AttachmentType: String, Codable {
+public enum AttachmentType: RawRepresentable, Codable, Equatable {
     /// An attachment type.
-    case unknown, image, imgur, giphy, video, youtube, product, file, link
+    case unknown
+    case custom(raw: String)
```

Comment:
I think it's a specific custom type, not a raw something. Could you please rename the parameter:
```swift
case .custom(type: String)
```